src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -28,14 +28,15 @@ public class DockerExecutor
         // Exception filter to exclude non-retriable errors from the docker daemon when attempting to pull images.
         private static readonly Func<DockerApiException, bool> IsNotAuthFailure = e =>
             // Immediately fail calls with either 'Unauthorized' or 'Forbidden' status codes
-            !(e.StatusCode == System.Net.HttpStatusCode.Unauthorized || e.StatusCode == System.Net.HttpStatusCode.Forbidden ||
+            !(e.StatusCode == HttpStatusCode.Unauthorized || e.StatusCode == HttpStatusCode.Forbidden ||
                 // Immediately fail calls with a status code of 'InternalServerError' and
                 // an HTTP body consisting of a JSON object with a single string property named "message" where the content is a colon-delimited string of three parts:
-                // a user-readable description of the failure, the status code from the remote registry server, and the textual description of that status code.
-                // Note that complete validation of the structure of the error report object is not performed. If a one word part after the first colon (and before any
-                // second colon) isn't found, this isn't the failure we are looking for and we'll assume it's retriable.
-                (e.StatusCode == System.Net.HttpStatusCode.InternalServerError &&
-                    new[] { "unauthorized", "forbidden" }.Contains(e.ResponseBody?.Split(':').Skip(1).FirstOrDefault()?.Trim().ToLowerInvariant() ?? string.Empty)));
+                // a user-readable rendition/description of the failure, the status code from the remote registry server, and the textual description of that status code
+                // (often with a web link). Note that complete validation of the structure of the error report object is not performed.
+                (e.StatusCode == HttpStatusCode.InternalServerError && (
+                    (e.ResponseBody?.Contains(": unauthorized:") ?? false) ||
```

Comment:
seems more future proof to just check if the responsebody contains "unauthorized" or "forbidden" in a case insensitive way